consumer.go

```diff
@@ -88,6 +92,56 @@ func (m *Msg) Data() []byte {
 	return m.msg.Data
 }
 
+// Msg.DataDeserialized - get message's deserialized data.
+func (m *Msg) DataDeserialized() (any, error) {
```

Comment:
Created a new function here to retrieve deserialized data. I would happy to change the name or change to different appoach to  retrieve deserialized data.